validator/accounts/status.go

```diff
@@ -0,0 +1,235 @@
+	}
+	statuses, err := FetchAccountStatuses(
+		ctx, ethpb.NewBeaconNodeValidatorClient(conn), pubkeys)
+	if e := conn.Close(); e != nil {
+		log.WithError(e).Error("Could not close connection to beacon node")
+	}
+	if err != nil {
+		return errors.Wrap(err, "Could not fetch account statuses from the beacon node")
+	}
+	printStatuses(statuses)
+	return nil
+}
+
+func constructDialOptions(
+	maxCallRecvMsgSize int,
+	withCert string,
+	grpcHeaders []string,
+	grpcRetries uint) ([]grpc.DialOption, error) {
+	var transportSecurity grpc.DialOption
+	if withCert != "" {
+		creds, err := credentials.NewClientTLSFromFile(withCert, "")
+		if err != nil {
+			return nil, errors.Wrapf(err, "Could not get valid credentials: %v", err)
+		}
+		transportSecurity = grpc.WithTransportCredentials(creds)
+	} else {
+		transportSecurity = grpc.WithInsecure()
+		log.Warn("You are using an insecure gRPC connection! Please provide a certificate and key to use a secure connection.")
+	}
+
+	if maxCallRecvMsgSize == 0 {
+		maxCallRecvMsgSize = 10 * 5 << 20 // Default 50Mb
+	}
+
+	md := make(metadata.MD)
+	for _, hdr := range grpcHeaders {
+		if hdr != "" {
+			ss := strings.Split(hdr, "=")
+			if len(ss) != 2 {
+				log.Warnf("Incorrect gRPC header flag format. Skipping %v", hdr)
+				continue
+			}
+			md.Set(ss[0], ss[1])
+		}
+	}
+
+	return []grpc.DialOption{
+		transportSecurity,
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(maxCallRecvMsgSize),
+			grpc_retry.WithMax(grpcRetries),
+			grpc.Header(&md),
+		),
+		grpc.WithStatsHandler(&ocgrpc.ClientHandler{}),
+		grpc.WithBlock(),
+		grpc.WithTimeout(
+			10 * time.Second /* Block for 10 seconds to see if we can connect to beacon node */),
+	}, nil
+}
+
+// FetchAccountStatuses fetches validator statuses from the BeaconNodeValidatorClient
+// for each validator public key.
+func FetchAccountStatuses(
+	ctx context.Context,
+	beaconNodeRPCProvider ethpb.BeaconNodeValidatorClient,
+	pubkeys [][]byte) ([]ValidatorStatusMetadata, error) {
+	ctx, span := trace.StartSpan(ctx, "accounts.FetchAccountStatuses")
+	defer span.End()
+
+	errorChannel := make(chan error, MaxRequestLimit)
+	statusChannel := make(chan []ValidatorStatusMetadata, MaxRequestLimit)
+	// Launch routines to fetch statuses.
+	i, numBatches := 0, 0
+	for ; i+MaxRequestKeys < len(pubkeys); i += MaxRequestKeys {
+		go fetchValidatorStatus(
+			ctx, beaconNodeRPCProvider, pubkeys[i:i+MaxRequestKeys], statusChannel, errorChannel)
+		numBatches++
+	}
+	if i < len(pubkeys) {
+		go fetchValidatorStatus(
+			ctx, beaconNodeRPCProvider, pubkeys[i:], statusChannel, errorChannel)
+		numBatches++
+	}
+	// Wait for fetch routines to finish.
+	var err error
+	allStatuses := make([][]ValidatorStatusMetadata, 0, numBatches)
+	for i := 0; i < numBatches; i++ {
+		select {
+		case statuses := <-statusChannel:
+			allStatuses = append(allStatuses, statuses)
+		case e := <-errorChannel:
+			err = e
+			log.Warnln(err)
+		}
```

Comment:
```suggestion
		}
		case <-ctx.Done():
		  return nil, ctx.Err()
```

Please handle cancelled / timed out context.